core/trino-main/src/main/java/io/trino/operator/DriverFactory.java

```diff
@@ -33,12 +33,13 @@
     private final int pipelineId;
     private final boolean inputDriver;
     private final boolean outputDriver;
+    @GuardedBy("this")
     private final List<OperatorFactory> operatorFactories;
     private final Optional<PlanNodeId> sourceId;
     private final OptionalInt driverInstances;
 
-    @GuardedBy("this")
-    private boolean noMoreDrivers;
+    // must synchronize between createDriver() and noMoreDrivers(), but isNoMoreDrivers() is safe without synchronizing
+    private volatile boolean noMoreDrivers;
```

Comment:
I would leave the `@GuardedBy`, and add a supress warning (with comment) on `isNoMoreDrivers`

Reply:
Ok, will change that in the latest revision.